### DISCUSSION TITLE
Trained CKPT - always same results

### DISCUSSION THREAD

**Discussion by akilmaf** [TheLastBen/fast-stable-diffusion#538] [Q&A] [answered]
Hi there,

First of all thank you for your contribution. I learned a lot and still learning. My problem is that after preparing everything and training on the pictures (23 photos, close shot) I was finally able to run the ckpt and generate new visuals. But every time I click generate, the result is always almost the same photo I uploaded while training. Same face & cloth. It acts like copying my photo and pasting after generating.

I created 3 ckpt files (spent 2 days...) so far with different colab tutorials. None of them worked visually. Where do I make mistake?

Thank you in advance.

**Comment by MasterDenis** [comment]
Hi, did you use regularization images? ("contains faces"),
Also How much was the number in your text encoder settings?

**Reply by akilmaf** [reply]
Hi,

- For regularization images, I could not find a place to upload them. Maybe the interface has been updated? Anyways I downloaded regularization images samples from this link, found them in a tutorial video, https://github.com/aitrepreneur/REGULARIZATION-IMAGES-SD. Where shall I put them? Feels like something is missing

- If you mean "Train_text_encoder_for" it was 100. Did not touch it.

**Reply by ykurilov** [reply]
Train_text_encoder_for when it's 100, I have the same result. Try to lower it to 50. That's what I'm trying to do now. I've also tried to make it 10% and the result was ok.

**Reply by akilmaf** [reply]
Yeah %50 works just fine. What I realised is that there should be at least 50 photos taken in different places & backgrounds. This way the training will be much better.

**Comment by TheLastBen** [comment, marked as answer]
Hi,
What did you name your instance images ?

**Reply by akilmaf** [reply]
I named them as example, example 1, example 2 etc..... 
In another attempt I used the default naming format which is example, example (1), example (2)

Is there a way to share my ckpt file details so you could see where the problem lies or if there is a problem?

If it helps: The photos I uploaded were taken in the same place, different angles, then in another training attempt I added additional photos taken somewhere else. Same result.. 

It feels like it does not recognize the face part so the whole photo remains the same with minor changes. At most, face changes a little bit but the location of the face, stance, background etc. remains the same in the generated image..

**Reply by TheLastBen** [reply]
set the text encoder to 20% and retrain from scratch and don't use the "contains_faces" option

**Reply by akilmaf** [reply]
Was just doing it now (text encoder was %40) . But contains_faces was on for the first time compared to my old attempts.

Is there an update on fast stable page? Because I can't see a line where I could upload regularization images. 

Also, previously, there was a line where you would name your prompt to trigger it. like "photo of asd" etc. 

Another one, "Total Steps = Number of Instance images * 200" section was like this before : "Number of Instance images * 100" Now we need to multiply it with 200. For 23 photos it was 2300 steps before. Now according to this one it should be 4600 steps. I tried both, the result was the same. Is it all experimental? 

Lastly, if I get the same bad result what would be the reason?

**Reply by TheLastBen** [reply]
Don't use  "contains_faces", set the steps to 4500 and the text encoder to 50%, this if you're using 23 input images

**Reply by akilmaf** [reply]
Update:
Despite the error below, managed to use the trained file on AUTOMATIC1111 colab. Results are slightly better compared to my complaint. Usually getting 2 characters, duplicated. 

**Interestingly, I get "uploaded photo" related results only by adding "akilmaf" in the prompt. I never added a prompt during training as there is no line to add it. The name of the ckpt file is Test_2.ckpt .** 

Will update this post if I get the desired results
----------------


Will try without "contains_faces" in my next attempt. Current one finished. Got this error after clicking Test The Trained Model:

[!] Not using xformers memory efficient attention.
Patching transformers to fix kwargs errors.
Dreambooth API layer loaded
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
making attention of type 'vanilla' with 512 in_channels
Working with z of shape (1, 4, 32, 32) = 4096 dimensions.
making attention of type 'vanilla' with 512 in_channels
Loading weights [afb7aa7a] from /content/gdrive/MyDrive/Fast-Dreambooth/Sessions/Test_2/Test_2.ckpt
Applying cross attention optimization (Doggettx).
Model loaded.
Traceback (most recent call last):
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 204, in <module>
    webui()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 147, in webui
    initialize()
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/webui.py", line 85, in initialize
    shared.opts.onchange("sd_vae_as_default", wrap_queued_call(lambda: modules.sd_vae.reload_vae_weights()), call=False)
  File "/content/gdrive/MyDrive/sd/stable-diffusion-webui/modules/shared.py", line 470, in onchange
    item.onchange = func
AttributeError: 'NoneType' object has no attribute 'onchange'

**Reply by akilmaf** [reply]
> Don't use "contains_faces", set the steps to 4500 and the text encoder to 50%, this if you're using 23 input images

Just trained ckpt file following this suggestion. Now it works fine. Sometimes I get results close to the faces in uploaded photos. 

Thanks a lot for the help. Now it is time to learn more :) 

Tip: If you use the name of the picture in the prompt you won't get creative results. "Person" works the best for me.

**Reply by TheLastBen** [reply]
use this prompt or similar : 

___________, Still frame , cinematic, 40mm f/2.8, remastered, 1990s film, 4k uhd, closeup, ultra detailed, highly intricate

Negative prompt: painting, 3d, low quality

Steps: 50, Sampler: Euler a, CFG scale: 7, Size: 512x768, Denoising strength: 0.7, First pass size: 0x0
use highres.fix

**Reply by akilmaf** [reply]
________, brings almost same uploaded pictures result. So using "person" , "man" works better. For me at least.

**Reply by nawnie** [reply]
Dead post but since ive seen TheLastBen use this example before "____, " means replace with your subject so from Joe's penny's guide it would be filled in xzx still frame... of a dog, tags.....

**Reply by brian-tam** [reply]
Hello Akilmaf,

How do you use "person" and "man"? 
I couldn't find the option to change the class and instance name

**Reply by nawnie** [reply]
this one uses a different system then shivam, you name your files what you want the instance to be. i have been toying with using file names as my instance prompts with good results, others have tried this as well. here is a discussion where i'm outlining what i'm doing, no examples just yet tho. working on a comparison project now.

https://github.com/TheLastBen/fast-stable-diffusion/discussions/427

**Reply by brian-tam** [reply]
Thank you nawnie. Does naming my files as instance prompts train the "person"/"man" class in the model?

**Reply by nawnie** [reply]
for all intense and purposes class does not exist, in this repo and only subject is defined. 

class is a term used when using images to compare the new subject too. so if you class something a man you show it many pictures of men so it can then compare what in your subject picture is similar to class this is why it was recommended you take different poses in different backgrounds so both class and your images have those variables changing each time so only what was unique about the subject needed to be recorded, this however uses your caption (filename) to replace the caption in the in the SD data set. 

if you need to use real names you are going to increase the need for training time, one thing i mentioned yesterday is when using the name whitney for a friend i got whitney houston with whitney my friends hair. we can counter this by using long unique prompts for file names, but then each word will call to that subject, so say my subject images have one labeled, whitney a petite girl wearing glasses blows out her birthday candles.

if i then try to do a prompt whitney drives her car, well whitney houston is very well represented in the sd model so ill get a picture of her, but if i then type whitney drives her car, wearing glasses on her birthday. and i'm more likely to get my subject image.
** if i use more commo spelling whittney this effect is less pronounced but can still blend with the original info 

TL:DR there is no class and i extended my reply to your other post here, ie dont use real names if you do use the most common spelling of it.

**Reply by brian-tam** [reply]
ty for the detailed reply nawnie! It was very helpful. I tried to make the filename as unique as I could like (tttwwwhhhbbb.jpg). I typed prompt like photo portrait of tttwwwhhhbbb, greek god, white hair, (and added some adjective), but it generates a similar style and the generated face does not look like me.  Do you why is it the case?

**Comment by TheLastBen** [comment]
make sure you check the box "update_repo"

**Reply by akilmaf** [reply]
> make sure you check the box "update_repo"

yeah it was checked.
What do you think that causes same photo generation? the generated images always come with the buildings behind me in the original photos. Even though I had other photos with blank background

**Comment by nawnie** [comment]
What are your settings for dreambooth, also have you tried adjusting the
cfg scale at bottom of ui or using (((name))).
Each parentheses adds weight to what's in it.

**Reply by brian-tam** [reply]
all my settings are default, I haven't use cfg scale and parentheses in my prompt...

**Reply by nawnie** [reply]
give these a try and it will help, now for your settings, is the text encoder set to 100 or 0? give it a try at 35

**Reply by brian-tam** [reply]
I used 100 and 0 for the text encoder.
Now I just tried 35, the result is much better! Thank you so much! But the style isn't showing what the prompt is requesting...
For example, if I include greek god in the prompt, it does not show the style of a greek god. Do I have to make the number of text encoder set bigger? like 40-50?

**Reply by TheLastBen** [reply]
if you have a hard time applying style, reduce the text_encoder % when you retrain from scratch

**Reply by nawnie** [reply]
encoder is lower for style info about the encoder is right about where you type in the amount, for a person you want about 40-50, to train a style you want about 10-15, i said 35 for a little extra flexability, 

your prompt may be the issue here so i found you one on Lexica

portrait of henry _____ as a heavenly god, full body, muscular, fantasy, intricate, elegant, highly detailed, digital painting, artstation, concept art, matte, sharp focus, illustration, art by artgerm and greg rutkowski and alphonse mucha 

a simpler adjustment greek god isnt a style its a title, using an artist name that makes art in the style you want helps alot but you could do, 

A detailed painting of ____ as a god, ((greek roman art style painting))


also a few more steps with no enocder may be needed if you are not able to leave a photo realistic style, setting the cfg scale to 6 or 6.5 should help, and i know 0 about the samplers, but i like kpm2a ++m for non real art, but try them all ddim works really well too

**Reply by brian-tam** [reply]
Sure! It looks like changing the text_encoder is essential to the style. I’ll try again. I got another question. I uploaded the trained model (my session) to hugging face, and loaded it with my api key, created the new session for training, but it asked me if I wanna overwrite an existing model or train a new model. But how do I fine tune a trained model? I would like to train the model with more data sets. Thank you!!

**Reply by nawnie** [reply]
Working on a very detailed guide with examples i hope to be up next week, also on my fork im about to try and put a 2 or 3 click dreambooth, bassicly you tell me how many subjects and ill have an option to use a recommended settings

will be LastBens instructions and if my test continue well ill include mine as a preset as well if it works well once Bens 2.0 is out ill attempt to merge or just ask as a separate easyBooth

**Reply by nawnie** [reply]
load the same session make sure resume is on and type no to resume or yes to start over

when i say to train a style i mean to train something like art in the style of Davinci. a person at 10% is doable with alot of extra steps, just a rule of thumb it seems no matter how many images at about 1k steps the encoder is done, i could be quitting to early in these trials but ill find out soon.

so maybe try training it for 100% for 350 steps then train until it looks good to you start at 200 per image

**Reply by brian-tam** [reply]
Thanks for the tip nawnie. This is very helpful! I just tried 100% for 350 steps, it produces promising results! If I overtrained my model at some point, do I have to start over to train the original model again? I now know how to fine-tune the model by using resume training. But if I overtrained it, what should I do? Thank you so much!!

**Reply by brian-tam** [reply]
I have 151 pictures in total, if I am training it at 100%, should I still use 200 steps per image?
